github/branches/validate_compliance.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+from typing import List
+
+from .retrieve_github_data import RepoBranchProtections, BranchProtectionRule
+from ..helpers import Criteria
+
+
+# define the criteria we care about. Identify each critera with a string that will
+# appear in the results.
+required_criteria = [
+    Criteria("SOGH001", "rules", "active rules"),
+    Criteria("SOGH001b", "admins", "admins restricted"),
+]
+optional_criteria = [
+    Criteria("SOGH001c", "commiters", "allowed commiters configured"),
+    # "commit signing",  # may not be knowable
+]
+warning_criteria = [
+    Criteria("SOGH001d", "conflicts", "Conflict in Protection Rules"),
+]
+
+
+def find_applicable_rules(
```

Comment:
Would love to see a doctest for this func as well

Reply:
coming post refactor